### PR TITLE
Add troubleshooting tips to prevent ProGuard over optimization

### PR DESCRIPTION
**Pull request by vixentael**
As discovered by @dodikk at https://github.com/cossacklabs/themis/issues/716, sometimes ProGuard/R8 can over-optimize builds and strip Themis classes. Add a line into docs just in case our users will get this issue.

Related https://github.com/cossacklabs/themis/issues/716#issuecomment-705243706, https://github.com/dodikk/themis-xamarin-prototype/pull/10/.

### CODE REVIEW

File: themis/languages/java/installation-android.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ title:  Installation for Android
 
 # Installing JavaThemis for Android development
 
-JavaThemis for Android is [available on _JFrog Bintray_ via **Maven**](https://bintray.com/cossacklabs/maven/themis).
+JavaThemis for Android is [available on JFrog Bintray via **JCenter** and **Maven**](https://bintray.com/cossacklabs/maven/themis).
```
Comment (ilammy) [inline themis/languages/java/installation-android.md:8]: Maven is a tool. Bintray and JCenter are repositories.
```suggestion
JavaThemis for Android is [available on JCenter and Bintray via **Maven**](https://bintray.com/cossacklabs/maven/themis).
```

File: themis/languages/java/installation-android.md [hunk 2]
```diff
@@ -35,6 +35,33 @@ If you experience difficulties with jcenter repository, try using maven instead:
 
 Once JavaThemis is installed, you can [try out examples on your machine](../examples/).
 
+
+## Configuring ProGuard / R8 rules
+
+Sometimes when working with Android projects in `Release` configuration, you might get runtime error, like:
```
Comment (ilammy) [inline themis/languages/java/installation-android.md:41]: This section is about a particular runtime error, not about dealing with runtime errors in general. I'd rephrase the sentence a bit to suggest that what follows is an example. Also, clauses usually don't need commas.

```suggestion
Sometimes when working with Android projects in `Release` configuration you might get the following runtime error:
```

File: themis/languages/java/installation-android.md [hunk 3]
```diff
@@ -35,6 +35,33 @@ If you experience difficulties with jcenter repository, try using maven instead:
...
+```
+Java.Lang.IncompatibleClassChangeError: no non-static method "Lcom/cossacklabs/themis/SecureCellSeal;.decrypt([B[B)[B"
+[orion.mobile]   at Java.Interop.JniEnvironment+InstanceMethods.GetMethodID (Java.Interop.JniObjectReference type, System.String name, System.String signature) [0x0005b] in <42d2b7086f0a46efb99253c5db1ecca9>:0 
+[orion.mobile]   at Android.Runtime.JNIEnv.GetMethodID (System.IntPtr kls, System.String name, System.String signature) [0x00007] in <3080427739614e60a939a88bf3f838d5>:0 
+[orion.mobile]   at Com.Cossacklabs.Themis.SecureCell+ISealInvoker.Decrypt (System.Byte[] p0, System.Byte[] p1) [0x00017] in <cd618986d1ce4194b63cdd3366dad291>:0 
+[orion.mobile]   at Themis.Droid.CellSealDroid.UnwrapData (Themis.ISecureCellData cipherTextData, System.Byte[] context) [0x0007e] in <a492e7118e094c3296442a386fe5d80e>:0 
+[orion.mobile]    --- End of inner exception stack trace --
+```
+
+It is caused by "over-optimizing" a build by ProGuard / R8, which strips native Java classes from a build.
```
Comment (ilammy) [inline themis/languages/java/installation-android.md:52]: Rephrased a bit so that the "blame" falls on ProGuard, rather then the developers who "over-optimized their build".

```suggestion
It is caused by ProGuard / R8 "over-optimizing" a build, stripping native Java classes as a result.
```

File: themis/languages/java/installation-android.md [hunk 4]
```diff
@@ -35,6 +35,33 @@ If you experience difficulties with jcenter repository, try using maven instead:
...
+[orion.mobile]   at Java.Interop.JniEnvironment+InstanceMethods.GetMethodID (Java.Interop.JniObjectReference type, System.String name, System.String signature) [0x0005b] in <42d2b7086f0a46efb99253c5db1ecca9>:0 
+[orion.mobile]   at Android.Runtime.JNIEnv.GetMethodID (System.IntPtr kls, System.String name, System.String signature) [0x00007] in <3080427739614e60a939a88bf3f838d5>:0 
+[orion.mobile]   at Com.Cossacklabs.Themis.SecureCell+ISealInvoker.Decrypt (System.Byte[] p0, System.Byte[] p1) [0x00017] in <cd618986d1ce4194b63cdd3366dad291>:0 
+[orion.mobile]   at Themis.Droid.CellSealDroid.UnwrapData (Themis.ISecureCellData cipherTextData, System.Byte[] context) [0x0007e] in <a492e7118e094c3296442a386fe5d80e>:0 
+[orion.mobile]    --- End of inner exception stack trace --
+```
+
+It is caused by "over-optimizing" a build by ProGuard / R8, which strips native Java classes from a build. 
+
+An easy solution is to add `proguard.cfg` file to the Android app project with the following lines:
```
Comment (ilammy) [inline themis/languages/java/installation-android.md:54]: A bit of a personal touch:
```suggestion
An easy solution is to add `proguard.cfg` file to your Android app project with the following lines:
```

File: themis/languages/java/installation-android.md [hunk 5]
```diff
@@ -35,6 +35,33 @@ If you experience difficulties with jcenter repository, try using maven instead:
...
+
+```
+-keep class com.cossacklabs.themis.**
+-keep class com.cossacklabs.themis.** {*;}
+```
+
+Make sure that the project has a `ProguardConfiguration` build step.
```
Comment (ilammy) [inline themis/languages/java/installation-android.md:61]: A bit more...
```suggestion
Make sure to enable the `ProguardConfiguration` build step after that.
```

File: themis/languages/java/installation-android.md [hunk 6]
```diff
@@ -35,6 +35,33 @@ If you experience difficulties with jcenter repository, try using maven instead:
...
+
+Please refer to [the issue #716](https://github.com/cossacklabs/themis/issues/716) for details, and to the [example application made by @dodikk](https://github.com/dodikk/themis-xamarin-prototype/pull/10) who has discovered and troubleshoot the issue.
```
Comment (ilammy) [inline themis/languages/java/installation-android.md:63]: I believe this verb should be inflected like this here:
```suggestion
Please refer to [the issue #716](https://github.com/cossacklabs/themis/issues/716) for details, and to the [example application made by @dodikk](https://github.com/dodikk/themis-xamarin-prototype/pull/10) who has discovered and troubleshot the issue.
```

**Review by ilammy — APPROVED** [review]
Yay, proofreading!

File: themis/languages/java/installation-android.md (hunk 1 above)
Comment (vixentael) [reply themis/languages/java/installation-android.md]: thank you, I'm being constantly confused by this

File: themis/languages/java/installation-android.md (hunk 4 above)
Comment (vixentael) [reply themis/languages/java/installation-android.md]: As I understood, there're certain Proguard "default" configurations on Release mode, which users might tweak with additional settings.

**Review by vixentael — COMMENTED** [review]
Great suggestions, thank you!
